### PR TITLE
Add AdditionalUsageInfo function

### PR DESCRIPTION
**Pull request by samwestmoreland**
Add an AdditionalUsageInfo function as part of change to make alias information visible in the usage info (goes with https://github.com/thought-machine/please/pull/2035 and https://github.com/thought-machine/go-flags/pull/5)

### CODE REVIEW

File: flags/flags.go [hunk 1]
```diff
@@ -16,14 +17,21 @@ import (
 // A CompletionHandler is the type of function that our flags library uses to handle completions.
 type CompletionHandler func(parser *flags.Parser, items []flags.Completion)
 
+// AdditionalUsageInfo is the type of function that seeks out auxiliary options and adds them
+// to the end of usage info
+type AdditionalUsageInfo func(parser *flags.Parser, wr *bufio.Writer)
```
Comment (peterebden) [inline flags/flags.go]: I think this should be `func(parser *flags.Parser, wr io.Writer)` - the callback shouldn't specifically care what kind of writer it is.

**Review by peterebden — COMMENTED** [review]
Thanks Sam!

File: flags/flags.go (hunk 1 above)
Comment (samwestmoreland) [reply flags/flags.go]: Fair point! Thanks :)

**Review by peterebden — APPROVED** [review]
Looks good, thanks!

This is a breaking change so we should do a new major version for it.

**Pull request merged**